lib/ts/fetch.ts

```diff
@@ -722,9 +723,11 @@ async function setAuthorizationHeaderIfRequired(clonedHeaders: Headers, addRefre
     const refreshToken = await getTokenForHeaderAuth("refresh");
 
     // We don't always need the refresh token because that's only required by the refresh call
-    // Still, we only add the Authorization header if both are present, because we are planning to add an option to expose the
-    // access token to the frontend while using cookie based auth - so that users can get the access token to use
-    if (accessToken !== undefined && refreshToken !== undefined) {
+    // Still, we only add the access token to Authorization header if both are present, because we are planning to add an option to expose the
```

Comment:
please add unit test for this in which you manually remove st-access-token from frontend cookie and then call the attemptRefresh function which should return true and add new tokens to frontend cookies